### PR TITLE
[#691] test :fix flaky test CoordinatorMetricsTest#testCoordinatorMetrics

### PR DESCRIPTION
**Pull request by smallzhongfeng**
### What changes were proposed in this pull request?
Do not start quotaManager when performing ut test of indicators.

### Why are the changes needed?
Fix: #691

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Fix ut.

### CODE REVIEW

**Comment by smallzhongfeng** [comment]
Please don't merge first. I will trigger GA more times. Thanks !

**Comment by smallzhongfeng** [comment]
I tried to find out why `testCoordinatorMetrics` in `CoordinatorMetricsTest` collected `app_num` in `QuotaManagerTest # testCheckQuotaMetrics`, but there are not many clues. At present, the simplest way is to filter out `app_num` in the indicators, do you have any good suggestions ? @jerqi @advancedxy @xianjingfeng 

The result as follow:
```
Test metrics => [
{"name":"running_app_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"total_quota_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"total_load_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"total_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"remote_storage_in_used_path1","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"total_candidates_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"total_app_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"app_num","labelNames":["user_name"],"labelValues":["user3"],"value":0.0,"timestampMs":null},
{"name":"app_num","labelNames":["user_name"],"labelValues":["user4"],"value":0.0,"timestampMs":null},
{"name":"unhealthy_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"total_access_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
{"name":"exclude_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null}]
```

**Comment by xianjingfeng** [comment]
> I tried to find out why `testCoordinatorMetrics` in `CoordinatorMetricsTest` collected `app_num` in `QuotaManagerTest # testCheckQuotaMetrics`, but there are not many clues. At present, the simplest way is to filter out `app_num` in the indicators, do you have any good suggestions ? @jerqi @advancedxy @xianjingfeng 
> 
> The result as follow:
> ```
> Test metrics => [
> {"name":"running_app_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_quota_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_load_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"remote_storage_in_used_path1","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_candidates_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_app_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"app_num","labelNames":["user_name"],"labelValues":["user3"],"value":0.0,"timestampMs":null},
> {"name":"app_num","labelNames":["user_name"],"labelValues":["user4"],"value":0.0,"timestampMs":null},
> {"name":"unhealthy_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_access_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"exclude_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null}]
> ```

I think we can remove this test case. it doesn't make sense.

**Comment by jerqi** [comment]
> I tried to find out why `testCoordinatorMetrics` in `CoordinatorMetricsTest` collected `app_num` in `QuotaManagerTest # testCheckQuotaMetrics`, but there are not many clues. At present, the simplest way is to filter out `app_num` in the indicators, do you have any good suggestions ? @jerqi @advancedxy @xianjingfeng
> 
> The result as follow:
> 
> ```
> Test metrics => [
> {"name":"running_app_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_quota_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_load_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"remote_storage_in_used_path1","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_candidates_denied_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_app_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"app_num","labelNames":["user_name"],"labelValues":["user3"],"value":0.0,"timestampMs":null},
> {"name":"app_num","labelNames":["user_name"],"labelValues":["user4"],"value":0.0,"timestampMs":null},
> {"name":"unhealthy_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"total_access_request","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null},
> {"name":"exclude_server_num","labelNames":[],"labelValues":[],"value":0.0,"timestampMs":null}]
> ```

We can filter them.

**Review by jerqi — APPROVED** [review]
LGTM, merged. thanks @smallzhongfeng

File: coordinator/src/test/java/org/apache/uniffle/coordinator/metric/CoordinatorMetricsTest.java [hunk 1]
```diff
@@ -89,7 +89,14 @@ public void testCoordinatorMetrics() throws Exception {
     ObjectMapper mapper = new ObjectMapper();
     JsonNode actualObj = mapper.readTree(content);
     assertEquals(2, actualObj.size());
-    assertEquals(10, actualObj.get("metrics").size());
+    int actualMetrics = 0;
+    for (JsonNode metrics : actualObj.get("metrics")) {
+      if (CoordinatorMetrics.APP_NUM_TO_USER.equals(metrics.get("name").textValue())) {
```
[the remark this answers is not in the document]
Comment (jerqi) [reply coordinator/src/test/java/org/apache/uniffle/coordinator/metric/CoordinatorMetricsTest.java:94]: @smallzhongfeng tried, but it doesn't matter. This test is so flaky that we should fix this issue asap. So we use the simplest to fix first.

[the remark this answers is not in the document]
Comment (smallzhongfeng) [reply coordinator/src/test/java/org/apache/uniffle/coordinator/metric/CoordinatorMetricsTest.java:94]: Thanks for your advice, I have tried this method, but it still doesn't work. I don't think it's the reason why the monitoring didn't clean up, but when I ran the GA, it seems like reading the latest indicators of other uts. I tried to modify the default `JETTY_ HTTP_ PORT` and `RPC_ SERVER_ PORT`, but still not effective.

**Pull request merged**